staging/src/k8s.io/api/README.md

```diff
@@ -1 +1,19 @@
-This repo is still in the experimental stage. Shortly it will contain the schema of the API that are served by the Kubernetes apiserver.
+# api
+
+Schema of the external API types that are served by the Kubernetes API server.
+
+## Purpose
+
+This library is the canonical location of the Kubernetes API definition. Most likely interaction with this repository is as a dependency of client-go.
+
+## Compatibility
+
+As a depenency of client-go, we provide the compatibility guarantees outlined at https://github.com/kubernetes/client-go#versioning, for the respective revisions of this repo in client-go's Godeps.json. It is in direct support of Kubernetes, so branches will track Kubernetes and be compatible with that repo. As we more cleanly separate the layers, we will review the compatibility guarantee.
```

Comment:
typo: dependency

> we provide the compatibility guarantees outlined at https://github.com/kubernetes/client-go#versioning

I don't think that's true... new exported fields in structs can break unkeyed constructions of objects, and we add those without bumping major versions in k8s.io/api (since we tag according to kubernetes releases)

I also think it is weird for a dependency of client-go to reference client-go :)

Reply:
This was added as per this comment suggestion https://github.com/kubernetes/kubernetes/pull/57759#discussion_r159374411

Any suggestions how to make it clear to a new user of the project what kind of compatibility guarantees there are?

Reply:
> new exported fields in structs can break unkeyed constructions of objects, and we add those without bumping major versions in k8s.io/api

another reason we alway bump our major semver digit on upstream releases.

Reply:
typo and feedback on compatibility promise is outstanding

Reply:
@liggitt we bump the major semver version of client-go on each 1.x.0 kube release. In contrast, we technically do not have major semver versions at all for k8s.io/api (the reason why we prefix the tags with `kubernetes-` instead of `v`). So I am sure what we should do with your comment.

Reply:
I wouldn't reference client-go in the api readme, and simply say "Branches track Kubernetes branches and are compatible with that repo."

Reply:
Ok, fine with me as well. @LiliC can you update? I think with the typos fixed we are ready to go then.